src/Riok.Mapperly/Descriptors/MappingBuilders/DictionaryMappingBuilder.cs

```diff
@@ -131,4 +134,25 @@ private static (ITypeSymbol, ITypeSymbol)? GetEnumerableKeyValueTypes(MappingBui
 
         return (enumeratedType.TypeArguments[0], enumeratedType.TypeArguments[1]);
     }
+
+    private static INamedTypeSymbol? GetExplicitIndexer(MappingBuilderContext ctx)
+    {
+        if (SymbolEqualityComparer.Default.Equals(ctx.Target.OriginalDefinition, ctx.Types.IDictionaryT))
+            return null;
+
+        // get typed version of IDictionay<>
+        ctx.Target.ImplementsGeneric(ctx.Types.IDictionaryT, out var typedInter);
+
+        var interfaceMethodSymbol = typedInter!.GetMembers(SetterIndexerPropertyName).OfType<IMethodSymbol>().SingleOrDefault(x => x.Parameters.Length == 2 && x.ReturnsVoid);
```

Comment:
Wouldn't it be more appropriate to use the `this` indexed property instead of the set method?

Reply:
I'd have to double check, but I think indexer methods become:
```C#
public V get_Item(K key);
public void set_Item(K key, V value);
```

Reply:
I think the indexer property is available under the name `this[]`.

Reply:
What's the advantage of getting the indexer over the setter? We only care about the setter.
With the indexer property we'd access `SetMethod` getting the exact same value as if we'd used "set_item".

```C#
// var targetIndexer = ctx.Target.GetMembers("this[]").FirstOrDefault(); // is null not sure why, might have to add types
var indexer = typedInter.GetMembers("this[]").First(); // get indexer with typed interface
var setter = indexer.SetMethod;
// here setter is the same as ctx.Target.GetMembers("set_Item").First();
var same = SymbolEqualityComparer.Default.Equals(indexer.SetMethod, setter); // true

```